tests/sources/test_confluence.py

```diff
@@ -403,6 +403,7 @@ async def test_remote_validation_when_space_keys_are_valid():
 
     with mock.patch("aiohttp.ClientSession.get", return_value=async_response):
         await source._remote_validation()
+    await source.close()
```

Comment:
Adding so many `await source.close()` is a signal to me, that we could make some changes to `create_source`, so that it can be used like:

```python
with create_source(ConfluenceDataSource) as source:
    assert source ...
```